definitions/npm/zustand_v5.x.x/flow_v0.229.x-/zustand_v5.x.x.js

```diff
@@ -0,0 +1,383 @@
+/* @flow */
+
+/**
+ * Reference:
+ * https://zustand.docs.pmnd.rs/apis/create-store
+ */
+
+/**
+ * Core api
+ */
+
+type zustand$SetStateInternal<S> =
+  & ((
+    partial: S | Partial<S> | (S => S | Partial<S>),
+    replace?: false
+  ) => void)
+  & ((
+    state: S | (S => S),
+    replace: true
+  ) => void)
+
+type zustand$StoreApi<S> = {
+  getInitialState: () => S,
+  getState: () => S,
+  setState: zustand$SetStateInternal<S>,
+}
+
+type zustand$CreateState<S> = (
+  setState: zustand$SetStateInternal<S>,
+  getState: () => S,
+  api: zustand$StoreApi<S>
+) => S
+
+declare module 'zustand/vanilla.js' {
+  declare export type StoreApi<S> = zustand$StoreApi<S>
+  declare export function createStore<S>(
+    createState: zustand$CreateState<S>
+  ): zustand$StoreApi<S>
+  declare export function createStore<S>(): (
+    createState: zustand$CreateState<S>
+  ) => zustand$StoreApi<S>
+}
+declare module 'zustand/vanilla.mjs' {
+  declare export type StoreApi<S> = zustand$StoreApi<S>
+  declare export function createStore<S>(
+    createState: zustand$CreateState<S>
+  ): zustand$StoreApi<S>
+  declare export function createStore<S>(): (
+    createState: zustand$CreateState<S>
+  ) => zustand$StoreApi<S>
```

Comment:
Maybe duplication could be avoided here by using [`$Exports`](https://flow.org/en/docs/types/utilities/#toc-exports)? Or use re-exports as done below in "Aliases"?